### PR TITLE
s2s backing contract

### PR DESCRIPTION
**Pull request by xiaoch05**

### CODE REVIEW

File: contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+    function changeDailyLimit(address token, uint amount) public onlyOwner  {
+        _changeDailyLimit(token, amount);
+    }
+
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    function pause() external onlyOwner {
```
Comment (HackFisher) [inline contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:67]: `onlyOperatorOrOwner` might be better?

File: contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+    function setMessagePalletIndex(uint32 index) external onlyOwner {
+        messagePalletIndex = index;
+    }
+
+    function changeDailyLimit(address token, uint amount) public onlyOwner  {
+        _changeDailyLimit(token, amount);
+    }
+
+    function unpause() external onlyOwner {
```
Comment (HackFisher) [inline contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:63]: Use `onlyOperatorOrOwner`?

**Comment by HackFisher** [comment]
Other parts LGTM.

File: contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol [hunk 3]
```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+import "@zeppelin-solidity-4.4.0/contracts/proxy/utils/Initializable.sol";
+import "@zeppelin-solidity-4.4.0/contracts/utils/math/SafeMath.sol";
+import "@darwinia/contracts-utils/contracts/DailyLimit.sol";
+import "@darwinia/contracts-utils/contracts/Ownable.sol";
+import "@darwinia/contracts-utils/contracts/Pausable.sol";
```
Comment (hujw77) [inline contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:7]: After #129, Utils is move to /wormhole/contracts/utils/

File: contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        // encode 
+        // this fee is needed to send to the sub<>sub bridge fund to pay relayer
+        bytes memory sendMessageCall = SubToSubBridge(DISPATCH_ENCODER).encode_send_message_dispatch_call(
+            messagePalletIndex,
+            laneId,
+            registerErc20,
+            fee);
+
+        // send s2s message
+        (bool success, ) = DISPATCH.call(sendMessageCall);
```
Comment (hujw77) [inline contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:123]: How the `fee` is deducted?

Comment (xiaoch05) [reply contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:123]: This fee is payed in the pallet parity-bridges-common. We don't need to set fee in this call.

Comment (hujw77) [reply contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:123]: It seems a force transfer?

Comment (hujw77) [reply contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:123]: And can anyone encode the call to deduct the fee by call `DISPATCH.call(sendMessageCall)`?

Comment (xiaoch05) [reply contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:123]: not force transfer, it's a inner not signed transfer https://github.com/paritytech/parity-bridges-common/blob/master/modules/messages/src/instant_payments.rs#L81

Comment (hujw77) [reply contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol:123]: ACK

**Comment by hujw77** [comment]
LGTM

**Review by HackFisher — APPROVED** [review]

**Pull request merged**